src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -944,8 +944,9 @@ private ImmutableHashSet<ISymbol> GetOrComputeGeneratedCodeSymbolsInTree(SyntaxT
         private ImmutableHashSet<ISymbol> ComputeGeneratedCodeSymbolsInTree(SyntaxTree tree, Compilation compilation, CancellationToken cancellationToken)
         {
             // PERF: Bail out early if file doesn't have "GeneratedCode" text.
-            var text = tree.GetText(cancellationToken).ToString();
-            if (!text.Contains("GeneratedCode"))
+            var walker = new GeneratedCodeTokenWalker();
+            walker.Visit(tree.GetRoot(cancellationToken));
```

Comment:
>Visit [](start = 19, length = 5)

Should we pay attention to ```cancellationToken``` while visiting?

Reply:
I lean towards 'no', since this is not something we did before. However, if you are not comfortable with that then I can add something.

Reply:
> I lean towards 'no', since this is not something we did before.

One might say that we did because ```GetText``` took the token as an argument.

---
In reply to: [371445741](https://github.com/dotnet/roslyn/pull/41209#discussion_r371445741) [](ancestors = 371445741)

Reply:
I will submit this change as a follow-up PR.

Reply:
➡️ #41250